src/main/java/io/vertx/proton/impl/ProtonSessionImpl.java

```diff
@@ -152,8 +153,12 @@ private String generateLinkName() {
 
   @Override
   public ProtonReceiver createReceiver(String address) {
-    // TODO: add options for configuring things like link name etc?
-    Receiver receiver = session.receiver(generateLinkName());
+      return createReceiver(address, new ProtonLinkOptions().setLinkName(generateLinkName()));
```

Comment:
I think maybe just passing a default 'blank' options here and checking the link name value for null (then generating one if it is) in the other method would be better, otherwise as further options are added you would also have to specify a link name to use any of them.